src/word.c

```diff
@@ -0,0 +1,67 @@
+#include "dictionary.h"
+#include <stdio.h>
+#include <strings.h>
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>	
+
+int num_punctuation = 22;
+char* punctuation_array[] = {"+",",",":"," ",".","-","'","&","!","?",":",";","#","~","=","/","$","£","^","\n","_","<",">"};
+
+// checks if punctuation exists within array
+int is_in_array(char* punctuation_array[], char* word) {
+    int i = 0;
+	for ( ; i < num_punctuation ; i++) {
+		if (strcmp(punctuation_array[i], word) == 0) {
+			//printf("comparing %s to %s", word, punctuation_array[i]);
+			return 1;	}
+	}
```

Comment:
This function is a very inefficient way to handle this. A much faster implementation is already provided for free for you in `dict_char_exists`.

Reply:
@sarikam1 

@dwahme how to change?

Reply:
@dwahme  do you know where i can find dict_char_exists? it's not in the dictionary.h or c file